internal/config_checks_rp2040.h

```diff
@@ -0,0 +1,113 @@
+#ifndef CONFIG_CHECK_RP2040_H
+#define CONFIG_CHECK_RP2040_H
+
+/** @ingroup hardware
+ * @page hardware_rp2040 Mozzi on RP2040 (Raspberry Pi Pico)
+ *
+ * port by Thomas Friedrichsmeier
+ *
+ * @section rp2040_status Port status and notes
+ * Compiles and runs using [this core](https://github.com/earlephilhower/arduino-pico). Can probably be ported to the Mbed core for RP2040, relatively easily,
+ * as it relies mostly on the RP2040 SDK API. Tested on a Pi Pico.
+ *
+ *   - This is a recent addition, implementation details may still change (currently just PWM driven by a timer; this may be worth changing to a DMA driven output)
+ *   - Wavetables and samples are not kept in progmem on this platform. While apparently speed (of the external flash) is not much of an issue, the data always seems to be copied into RAM, anyway.
+ *   - Note that getAudioInput() and mozziAnalogRead() return values in the RP2040's full ADC resolution of 0-4095 rather than AVR's 0-1023.
+ *   - twi_nonblock is not ported
+ *   - Code uses only one CPU core
+ *
+ * @section rp2040_output Output modes
+ * The following audio modes (see @ref MOZZI_AUDIO_MODE) are currently supported on this hardware:
+ *   - MOZZI_OUTPUT_EXTERNAL_TIMED
+ *   - MOZZI_OUTPUT_EXTERNAL_CUSTOM
+ *   - MOZZI_OUTPUT_PDM
```

Comment:
Is there really a PDM mode for the RP2040? Did not remember it and could not find it…

Reply:
Mixed it up with PWM. Fixed. (Although I always wanted to add a PDM mode for the RP2040, one of those days...)